src/components/SideBar/index.tsx

```diff
@@ -0,0 +1,171 @@
+'use client'
+
+import Link from "next/link";
+import Image from "next/image";
+import { usePathname, useRouter, useSearchParams } from 'next/navigation'
+import { useState } from "react";
+import {
+  ArrowRightFromLine,
+  DiscAlbum,
+  Flag,
+  Gauge,
+  Globe,
+  Guitar,
+  Home,
+  Library,
+  Moon,
+  Settings,
+  Sun,
+  User,
+  Users
+} from "lucide-react";
+import { parseClassName } from "@/utils/helpers";
+import SideBarHeader from "./SideBarHeader";
+import styles from "./sidebar.module.css";
+
+const navigation = [
+  {
+    links: [
+      {
+        href: "/home",
+        name: "Home",
+        icon: Home
+      },
+      {
+        href: "/community",
+        name: "Community",
+        icon: Globe,
+      },
+      {
+        href: "/dashboard",
+        name: "Dashboard",
+        icon: Gauge,
+      }
+    ]
+  },
+  {
+    separator: "Leaderboards",
+    links: [
+      {
+        href: "/player",
+        name: "Player",
+        icon: User
+      },
+      {
+        href: "/band",
+        name: "Band",
+        icon: Users,
+      },
+      {
+        href: "/instrument",
+        name: "Instrument",
+        icon: Guitar,
+      },
+      {
+        href: "/country",
+        name: "Country",
+        icon: Flag,
+      },
+      {
+        href: "/setlist",
+        name: "Setlist",
+        icon: Library,
+      },
+      {
+        href: "/source",
+        name: "Source",
+        icon: DiscAlbum,
+      }
+    ]
+  },
+  {
+    separator: true,
+    links: [
+      {
+        href: "/settings",
+        name: "Settings",
+        icon: Settings
+      }
+    ]
+  }
+]
+
+const languages = [
+  { name: "English", acronym: "en"},
+  { name: "Português", acronym: "pt"}
+]
+
+const release = {
+  image: "https://news.yarg.in/images/banners/v0-12-release.webp",
+  target: "#!",
+  text: "New release 0.12" 
+}
+
+const version = "v0.12.0"
+
+const SideBar: React.FC = () => {
+  const router = useRouter();
```

Comment:
const { push } = useRouter();

destructuring in this case, can be good be have a readable code.